File: 1.两数之和.js
/*
 * @lc app=leetcode.cn id=1 lang=javascript
 *
 * [1] 两数之和
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
  var map = new Map()
  var i = 0
  for (; i < nums. length; i++) {
    if (map.has(nums[i])) {
      return [map.get(nums[i]), i]
    } else {
      map.set(target - nums[i], i)
    }
  }
  return [-1, -1]
};
// @lc code=end
var nums = [2, 7, 11, 15]
var target = 9
var res = twoSum(nums, target)
console.log(res)

File: 1.两数之和.py
# pylint: disable=unused-wildcard-import
# pylint: disable=import-error

# @before-stub-for-debug-begin
from python3problem1 import *
from typing import *
# @before-stub-for-debug-end

#
# @lc app=leetcode.cn id=1 lang=python3
#
# [1] 两数之和
#

# @lc code=start
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map = {}
        for i, v in enumerate(nums):
            j = map.get(target - v)
            if j != None :
                return [j, i]
            else:
                map[v] = i
        return [-1, -1]
# @lc code=end

